libs/api/domains/education/src/lib/education.service.ts

```diff
@@ -24,7 +45,105 @@ export class EducationService {
   async downloadPdfLicense(
     nationalId: string,
     licenseId: string,
-  ): Promise<Response> {
-    return this.mmsApi.downloadLicensePDF(nationalId, licenseId)
+  ): Promise<string | null> {
+    const responseStream = await this.mmsApi.downloadLicensePDF(
+      nationalId,
+      licenseId,
+    )
+
+    return this.s3Service.uploadFileFromStream(responseStream, {
+      fileName: uuid(),
+      bucket: this.config.fileDownloadBucket,
+    })
+  }
+
+  async getFamily(nationalId: string) {
+    const family = await this.nationalRegistryApi.getMyFamily(nationalId)
+    return family.filter(
+      (familyMember) =>
+        nationalId === familyMember.Kennitala ||
+        (!['1', '2'].includes(familyMember.Kyn) &&
+          kennitala.info(nationalId).age < ADULT_AGE_LIMIT),
+    )
+  }
+
+  async getExamOverviews(nationalId: string): Promise<ExamOverview[]> {
+    const family = await this.getFamily(nationalId)
+    const examOverviews = await Promise.all(
+      family.map(async (familyMember) => {
+        const studentAssessment = await this.mmsApi.getStudentAssessment(
+          familyMember.Kennitala,
+        )
+        if (studentAssessment.einkunnir.length <= 0) {
+          return undefined
+        }
+
+        const examDates = flatten(
+          studentAssessment.einkunnir.map((einkunn) => [
+            einkunn.islenska?.dagsetning,
+            einkunn.enska?.dagsetning,
+            einkunn.staerdfraedi?.dagsetning,
+          ]),
+        ).filter(Boolean)
+
+        return {
+          nationalId: familyMember.Kennitala,
+          name: familyMember.Nafn,
+          isChild: nationalId !== familyMember.Kennitala,
+          organizationType: 'Menntamálastofnun',
+          organizationName: 'Samræmd Könnunarpróf',
+          yearInterval: getYearInterval(examDates),
+        }
+      }),
+    )
+    return examOverviews.filter(Boolean)
+  }
+
+  mapGrade(grade: GradeResult) {
+    return {
+      grade: grade.radeinkunn,
+      weight: grade.vaegi,
+    }
+  }
+
+  mapBaseGrade(grade: BaseGrade) {
+    return {
+      grade: grade.samtals.radeinkunn,
+      competence: grade.haefnieinkunn,
+      competenceStatus: grade.haefnieinkunnStada,
+      progressText: grade.framfaraTexti,
+    }
+  }
+
+  mapLanguageGrade(grade: LanguageGrade) {
+    return {
+      ...this.mapBaseGrade(grade),
+      readingGrade: this.mapGrade(grade.lesskilningur),
+      grammarGrade: this.mapGrade(grade.malnotkun),
+    }
+  }
+
+  mapMathGrade(grade: MathGrade) {
```

Comment:
Your mappings and `getFamily` are missing the return type. Thought we had strict typing.

Reply:
Good point, I will fix and check if I can turn the strict on

Reply:
On second thought, the type in `getFamily` is derived from the nationalRegistryApi as I never recreate the object.
And for the mapping functions, the `getExamResults` has a return type and there is no real gain from putting return types on the mapping functions. I just adds verbosity.

Reply:
aiit :+1: